packages/ckeditor5-engine/src/conversion/modelconsumable.ts

```diff
@@ -179,6 +179,30 @@ export default class ModelConsumable {
 		}
 	}
 
+	/**
+	 * Removes all consumable values from a given model item.
+	 *
+	 * It is equal to calling {@link module:engine/conversion/modelconsumable~ModelConsumable#consume `consume()`} for all consumable types
+	 * for a given model item.
+	 *
+	 * This is useful to prevent any further conversion for a given model item.
+	 *
+	 * @param item Model item, range or selection from which all consumables will be consumed.
+	 */
+	public consumeAll( item: Item | Selection | DocumentSelection | Range ): void {
```

Comment:
We could do it internally in `DowncastDispatcher#convertSelection()` but I thought that having a proper function would be just, well, a proper way to handle this, instead of implementing some internal magic (which I usually do, and I am not happy about it :)).

Initially, I thought that maybe `selection` event converters could, like, return `false` or stop the event and that would be used in `DowncastDispatcher#convertSelection()` to inform it that it should cancel the downcast process.

Honestly, I was a bit surprised that we don't have such method. Say that I have a converter for a custom, isolated widget and I have one function that takes care about the element and its attributes. Now I need to consume them manually. It's probably safer but more hassle. I thought we are doing something like this in tables.

I agree that it could lead to incorrect/lazy implementations of 3rd party features.

Reply:
> I agree that it could lead to incorrect/lazy implementations of 3rd party features.

Maybe we could mark it as internal to avoid such scenarios?

Reply:
I wonder if it could not be useful for integrators too? Maybe I can write a warning in the API that will explain how it will work and when you should use it?

Reply:
I can see use for this API for widgets like ToC that store the whole content of ToC in data but AFAIR it's discarded on upcast and generated from headings in content. Maybe we could have this method but we should notice there that you should remember that there could be other features that should handle their attributes etc (for example GHS).

Reply:
I just checked and ToC is just marking ToC content as CDATA (raw content) and just ignores it.

Reply:
Good point with GHS :thinking:. Okay, I will go with your suggestion and let's make it `@internal`.

Reply:
Okay, in the end, I decided to revert `consumeAll()` and apply the logic straight in `DowncastDispatcher`.